### ISSUE TITLE
Should retranslocation use Metabolic as well as Storage in DM similar to N

### ISSUE THREAD

**Issue by jbrider** [APSIMInitiative/ApsimX#3562]
In @2200 the conversation suggested that for N retranslocation - Storage was used first, and then Metabolic - but I don't think the code in GenericOrgan is working that way.

DM only uses Storage - so my issue is should it use Metabolic as well, and should that be the same precedence.

**Comment by HamishBrownPFR** [comment]
I am pretty sure that there are not any PMF models where generic organ is parameterised to put biomass into metabolic so at the moment it doesn't matter if Metabolic is not include in retranslocation calculations.  I suspect (although I am often wrong) that you could add metabolic into the calculation if you were needing it for Sorgham, without it effecting the function of existing models.

**Comment by jbrider** [comment]
I am looking at refactoring the retranslocation code, so I was wondering why N and DM were handled slightly differently before I started. 

Sorghum doesn't distinguish between structural and metabolic DM at the moment. It's all one pool at present - which I need to handle inside of GenericOrgan. 

So I will go ahead using N should be Storage first then Metabolic and DM doesn't use Metabolic.

**Comment by rcichota** [comment]
I was trying to use metabolic in generic organ (or simple leaf which used to inherited it). I might have implemented (or enabled) the metabolic N in those organs, but I did not go far (and that's maybe why DM hasn't it).  A major issue I found was how the pools are defined and whether there should any correspondence between them. The fractions of N are given by the N concentration, whereas DM has explicit fractions. Currently one can define MinimumN different than CriticalN (so have a metabolic N) but still have no metabolic DM; and that could be done also for storage. Should this be allowed? which pools should be linked?
